### PR TITLE
ENH: Output thickness, curvature, and sulcal depth files

### PR DESCRIPTION
**Pull request by ericfeczko**
This address issue #288. I'm running into some errors with the outputs due to some syntax issues that were introduced. Will be doing some testing with circle CI today to debug. Will also ensure pep8 compatibility.

### CODE REVIEW

File: smriprep/workflows/surfaces.py [hunk 1]
```diff
@@ -552,9 +567,17 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
+                                        ('sulc', 'in2'),
+                                        ('curv', 'in3')]),
+        (get_surfaces, whitesurf_list, [('smoothwm', 'in1'),
+                                        ('smoothwm', 'in2'),
+                                        ('smoothwm', 'in3')]),
+        (whitesurf_list, fscurv2funcgii, [('out', 'in_file')]),
+        (surfmorph_list, fscurv2funcgii, [('out', 'scalarcurv_file')]),
+        (fs2gii, allsurf_list, [('converted', 'in1')]),
+        (fscurv2funcgii, allsurf_list, [('converted', 'in2')]),
+        (allsurf_list, outputnode, [('out', 'surfaces')])
```
Comment (effigies) [inline smriprep/workflows/surfaces.py:580]: I think we're going to have trouble conflating `.surf.gii` files and `.shape.gii` files. I think we should probably not combine them and instead make a new `morphometrics` output. Something like:

```suggestion
        (fscurv2funcgii, outputnode, [('converted', 'morphometrics')]),
```

File: smriprep/workflows/surfaces.py [hunk 2]
```diff
@@ -534,7 +531,25 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
+    )
+    fscurv2funcgii = pe.MapNode(
+        fs.MRIsConvert(out_datatype="gii", to_scanner=True),
+        iterfield=["in_file", "scalarcurv_file"], name="fscurv2funcgii",
+    )
+    allsurf_list = pe.Node(
+        niu.Merge(7),
+        name="allsurf_list",
+        run_without_submitting=True,
+    )
```
Comment (effigies) [inline smriprep/workflows/surfaces.py:552]:
```suggestion
```

File: smriprep/workflows/surfaces.py [hunk 3]
```diff
@@ -534,7 +531,25 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
+    whitesurf_list = pe.Node(
+        niu.Merge(3, ravel_inputs=True),
+        name="whitesurf_list",
+        run_without_submitting=True,
+    )
+    fscurv2funcgii = pe.MapNode(
+        fs.MRIsConvert(out_datatype="gii", to_scanner=True),
+        iterfield=["in_file", "scalarcurv_file"], name="fscurv2funcgii",
+    )
```
Comment (effigies) [inline smriprep/workflows/surfaces.py:547]: We don't need to iterate if `in_file` is always the same.

```suggestion
    fscurv2funcgii = pe.MapNode(
        fs.MRIsConvert(out_datatype="gii"),
        iterfield=["scalarcurv_file"], name="fscurv2funcgii",
    )
```

File: smriprep/workflows/surfaces.py [hunk 4]
```diff
@@ -552,9 +567,17 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
-        (fs2gii, fix_surfs, [('converted', 'in_file')]),
-        (inputnode, fix_surfs, [('fsnative2t1w_xfm', 'transform_file')]),
-        (fix_surfs, outputnode, [('out_file', 'surfaces')]),
+        (get_surfaces, surfmorph_list, [('thickness', 'in1'),
+                                        ('sulc', 'in2'),
+                                        ('curv', 'in3')]),
+        (get_surfaces, whitesurf_list, [('smoothwm', 'in1'),
+                                        ('smoothwm', 'in2'),
+                                        ('smoothwm', 'in3')]),
+        (whitesurf_list, fscurv2funcgii, [('out', 'in_file')]),
```
Comment (effigies) [inline smriprep/workflows/surfaces.py:576]:
```suggestion
        (get_surfaces, fscurv2funcgii, [('smoothwm', 'in_file')]),
```

**Review by effigies — COMMENTED** [review]
If we restore all of the removed lines, we'll preserve the work done in #295. Then I think we can reduce this to a relatively small addition. We'll then need to go into `outputs.py` and add a couple nodes to write out the files correctly, mirroring what we do for `.surf.gii`:

https://github.com/nipreps/smriprep/blob/b29578ec45e51da63e5df6b47a8c88b02ae43acd/smriprep/workflows/outputs.py#L592-L601

File: smriprep/workflows/surfaces.py (hunk 3 above)
Comment (ericfeczko) [reply smriprep/workflows/surfaces.py:547]: won't the in_file vary by hemisphere? I'm a little confused by the interface, I'll admit :)

File: smriprep/workflows/surfaces.py (hunk 1 above)
Comment (ericfeczko) [reply smriprep/workflows/surfaces.py:580]: that makes a lot of sense -- I'll make sure to do the same moving forward when handling the cifti conversion

File: smriprep/workflows/surfaces.py (hunk 3 above)
Comment (effigies) [reply smriprep/workflows/surfaces.py:547]: Oh right. That might be the best way to do this after all.

**Comment by ericfeczko** [comment]
that sounds perfect -- I'm in meeting hell right now, but I can try to give this a shot tonight or tomorrow.

**Comment by effigies** [comment]
Here's an (untested) attempt: 857e0bd8370d967691dc11d9ca7b705e63efb394

You may need to add my fork to try it out:

```
git remote add effigies https://github.com/effigies/smriprep.git
git fetch effigies
git cherry-pick 857e0bd
```

**Comment by effigies** [comment]
What are the expected outputs? I guess `.dscalar.nii` or `.shape.gii`?

**Comment by mgxd** [comment]
Closed by #305

**Pull request closed without merging**